storage-proofs/post/src/fallback/compound.rs

```diff
@@ -65,10 +64,16 @@ impl<'a, Tree: 'static + MerkleTreeTrait>
 
             // 2. Inputs for verifying inclusion paths
             for n in 0..pub_params.challenge_count {
-                let challenge_index = ((partition_index * pub_params.sector_count + i)
-                    * pub_params.challenge_count
-                    + n) as u64;
-                let challenged_leaf_start = fallback::generate_leaf_challenge(
+                let challenge_index = if pub_params.is_winning {
+                    // Note that this generality is perhaps over-complicated and unnecessary
+                    // with the current parameterization.  To avoid complexity, the
+                    // challenge_index could be set to 'i' here.
+                    (partition_index * pub_params.sector_count + i) * pub_params.challenge_count + n
```

Comment:
Can we guard the simplifying assumption with an assertion then revert to the simple expression? That works for me.